### PR TITLE
Transactions queue documentation

### PR DESCRIPTION
**Pull request by Tbaut**
Provides a clear explanation of how Parity client manages the transactions, their verification, and their ordering.

### CODE REVIEW

File: Transactions-Queue.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+or
+- the nonce is `future` (there is a nonce gap)
+
+### Dropping conditions
+Once a transaction has been added to the queue, it might get dropped if:
+- any pre-verified condition has changed (nonce and balance change in the meantime)
+- the queue length limit is reached (configurable with the flag `--tx-queue-size`)
+- the per-sender max number of transactions is reached (configurable with the flag `--tx-queue-per-sender`)
+- memory limit was reached (configurable with the flag `--tx-queue-mem-limit`)
+
```
Comment (tomusdrw) [inline Transactions-Queue.md]: Another condition for drop: A different transaction with the same nonce, signed by the sender is replacing transaction that is already in the pool. For that to happen new `gas price` has to be at least 12.5% higher than the previous one.

File: Transactions-Queue.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Note also that local transactions will never get dropped to ease debugging.
+
+### Dropping logic
+Given the following transactions in the queue from a *same sender*:
```
Comment (tomusdrw) [inline Transactions-Queue.md]: a -> the?

File: Transactions-Queue.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- `drop` - means that transaction was added to the queue but got removed.
+
+### Accepting conditions
+
+As a transaction is received from the network, a verification is performed before adding it to the queue. The following criteria are checked:
+- the transaction's gas price is above the minimum set (configurable with the flag `--min-gas-price`).
+- the amount of gas used by the transaction is below the maximum set (configurable with the flag `--tx-queue-gas`).
+- the account sending the transaction has a sufficient balance.
+- the transaction's signature is correct.
+- the nonce is not in the *past*.
```
Comment (tomusdrw) [inline Transactions-Queue.md]: Another one: Byte length of RLP form of transaction does not exceed `300 * 1024` (or whatever is configured in the spec file).

File: Transactions-Queue.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+## Queue logic
+
+### Terminology
+- `rejected` - means that transaction will not be added in the queue.
+- `drop` - means that transaction was added to the queue but got removed.
+
+### Accepting conditions
+
+As a transaction is received from the network, a verification is performed before adding it to the queue. The following criteria are checked:
+- the transaction's gas price is above the minimum set (configurable with the flag `--min-gas-price`).
```
Comment (tomusdrw) [inline Transactions-Queue.md]: We should also mention and link to service transactions - special kind of zero gas price transactions that are allowed by on-chain managed contract.

File: Transactions-Queue.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+Note that a transaction would be included in the queue if it fulfills the above-mentioned criteria and:
+- the nonce is `current` (equals to state nonce + number of already pending transactions)
+or
+- the nonce is `future` (there is a nonce gap)
+
+### Dropping conditions
+Once a transaction has been added to the queue, it might get dropped if:
+- any pre-verified condition has changed (nonce and balance change in the meantime)
+- the queue length limit is reached (configurable with the flag `--tx-queue-size`)
```
[the remark this answers is not in the document]
Comment (Tbaut) [reply Transactions-Queue.md]: fixed

**Comment by Tbaut** [comment]
Thanks, I addressed all your comments.

File: Transactions-Queue.md [hunk 6]
```diff
@@ -7,34 +7,39 @@ Parity client is listening to the network for transactions in order to include t
...
-- `rejected` - means that transaction will not be added in the queue.
+- `rejected` - means that transaction will not be added to the queue.
 - `drop` - means that transaction was added to the queue but got removed.
 
 ### Accepting conditions
 
 As a transaction is received from the network, a verification is performed before adding it to the queue. The following criteria are checked:
-- the transaction's gas price is above the minimum set (configurable with the flag `--min-gas-price`).
+- the transaction's gas price is above the minimum set (configurable with the flag `--min-gas-price`) unless it is a [zero gas price transactions aka service transaction](https://wiki.parity.io/Permissioning.html#gas-price) allowed by an on-chain managed contract.
+- the transaction's gas price is lower than the current block gas limit.
```
Comment (tomusdrw) [inline Transactions-Queue.md]: transaction's gas, not gas price

**Comment by Tbaut** [comment]
I missed that.
Replaced by 
>the amount of gas used by the transaction is lower than the current block gas limit.

**Pull request merged**